packages/query-core/src/queriesObserver.ts

```diff
@@ -240,28 +240,14 @@ export class QueriesObserver<
           observer: match,
         })
       } else {
-        const existingObserver = this.#observers.find(
-          (o) => o.options.queryHash === defaultedOptions.queryHash,
-        )
         observers.push({
           defaultedQueryOptions: defaultedOptions,
-          observer:
-            existingObserver ??
-            new QueryObserver(this.#client, defaultedOptions),
+          observer: new QueryObserver(this.#client, defaultedOptions),
```

Comment:
I doubt that this is a correct fix, and I’d expect some tests to fail. This means that a re-render with the same queries would throw away observers and re-create them (see the `setQueries` function)

Reply:
yeah I think you’re right - it seems like this code doesn’t do anything. Good find 👍